samples/peripheral/radio_test/src/radio_test.c

```diff
@@ -478,6 +475,17 @@ static void radio_modulated_tx_carrier_duty_cycle(uint8_t mode, uint8_t txpower,
 
 #endif /* CONFIG_FEM */
 
+	const uint32_t total_time_per_payload = time_in_us_per_byte[mode] * total_payload_size;
+
+	/* Duty cycle = 100 * Time_on / (time_on + time_off),
+	 * we need to calculate "time_off" for delay.
+	 * In addition, the timer includes the "total_time_per_payload",
+	 * so we need to add this to the total timer cycle.
+	 */
+	uint32_t delay_time = total_time_per_payload +
```

Comment:
I checked why it isn't possible to have high duty cycle above (90% mostly). It is not possible because for some PHYs we use normal ramp-up time which is something about 130us for most of our SoC. When we set high duty cycle the radio off time is smaller than the radio rump.

Reply:
I enabled a fast ramp-up for all modes. Also I have limited the duty cycle range. Now it is 1-90%